js_modules/dagit/packages/core/src/runs/RunActionsMenu.tsx

```diff
@@ -83,12 +90,7 @@ export const RunActionsMenu: React.FC<{
               text={loading ? 'Loading Configuration...' : 'View Configuration...'}
               disabled={!runConfigYaml}
               icon="open_in_new"
-              onClick={() =>
-                showCustomAlert({
-                  title: 'Config',
-                  body: <HighlightedCodeBlock value={runConfigYaml || ''} language="yaml" />,
-                })
-              }
+              onClick={() => setVisibleDialog('config')}
```

Comment:
I think this looks good, especially because the component is already using visibleDialog state, but I've been sort of trying to push things toward the `showCustomAlert` model - in a perfect world I think we'd change the other dialogs in this file to work like `showCustomAlert`...

I like the `showCustomAlert` style better because the top level component is rendered once and detatched from the callsite, so the RunActionsMenu being re-rendered / unmounted / etc. can't blow away the dialog.

Reply:
Ah, interesting. I'm actually not really a fan of moving dialog control out of the component tree, and would prefer that we make sure that `Dialog` callsites guard against re-renders. We should talk about it. :)